app/models/event.rb

```diff
@@ -0,0 +1,9 @@
+class Event < ActiveRecord::Base
+  has_many :user_events
+  has_many :users, through: :user_events
+  belongs_to :location
+
+  accepts_nested_attributes_for :users, allow_destroy: true
```

Comment:
As I understand it's for nested form: when admin adds users to the speakers -- he also can destroy one of them (remove from the speakers of a certain event). Not sure. It's uncompleted part.